[bug]: Calendar module fails to build with libical 4.X

### Severity

Trivial

### Versions

23.3.0

### Components/Modules

res_calendar.so

### Operating Environment

Build OS: ArchLinux (Packages from 2026-05-26) within a clean chroot.

### Frequency of Occurrence

None

### Issue Description

Building asterisk's calendar functionality failed with libical 4.0.1. I attached the build output below.

### Relevant log output

```shell
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for wchar.h... yes
checking for minix/config.h... no
checking for sys/param.h... yes
checking for sys/select.h... yes
checking for sys/socket.h... yes
checking for utime.h... yes
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
checking whether char is unsigned... no
checking for uname... /usr/bin/uname
checking for g++... g++
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... -std=gnu++11
checking how to run the C preprocessor... gcc -E
checking how to run the C++ preprocessor... g++ -std=gnu++11 -E
checking whether g++ -std=gnu++11 supports C++11 features by default... yes
checking whether g++ -std=gnu++11 supports C++14 features by default... no
checking whether g++ -std=gnu++11 supports C++14 features with -std=gnu++14... yes
checking whether g++ -std=gnu++11 supports C++17 features by default... no
checking whether g++ -std=gnu++11 supports C++17 features with -std=gnu++17... yes
checking whether g++ -std=gnu++11 supports C++20 features by default... no
checking whether g++ -std=gnu++11 supports C++20 features with -std=gnu++20... yes
checking whether g++ -std=gnu++11 supports C++23 features by default... no
checking whether g++ -std=gnu++11 supports C++23 features with -std=gnu++23... yes
checking for a sed that does not truncate output... /usr/bin/sed
checking for egrep... grep -E
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for gawk... gawk
checking for a BSD-compatible install... /usr/bin/install -c
checking whether ln -s works... yes
checking for ranlib... ranlib
checking for GNU make... make
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for strip... strip
checking for ar... ar
checking for bison... /usr/bin/bison
checking for cmp... /usr/bin/cmp
checking for cat... /usr/bin/cat
checking for cut... /usr/bin/cut
checking for flex... /usr/bin/flex
checking for grep... (cached) /usr/bin/grep
checking for python2.7... no
checking for python2... no
checking for python... /usr/bin/python
checking for find... /usr/bin/find
checking for basename... /usr/bin/basename
checking for dirname... /usr/bin/dirname
checking for sh... /bin/sh
checking for ln... /usr/bin/ln
checking for doxygen... :
checking for dot... :
checking for wget... :
checking for curl... /usr/bin/curl
checking for xmllint... /usr/bin/xmllint
checking for xmlstarlet... no
checking for xml... no
checking for bash... /bin/sh
checking for git... :
checking for alembic... :
checking for bzip2... /usr/bin/bzip2
checking for tar... /usr/bin/tar
checking for patch... /usr/bin/patch
checking for sed... (cached) /usr/bin/sed
checking for nm... /usr/bin/nm
checking for realpath... /usr/bin/realpath
checking for ldconfig... /usr/bin/ldconfig
checking for sha1sum... /usr/bin/sha1sum
checking for openssl... /usr/bin/openssl
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for bison that supports parse-param... /usr/bin/bison
checking for soxmix... no
checking for md5... no
checking for md5sum... md5sum
checking for a sed that does not truncate output... (cached) /usr/bin/sed
checking for egrep -e... (cached) /usr/bin/grep -E
checking whether gcc is Clang... no
checking whether pthreads work with "-pthread" and "-lpthread"... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking for RAII support... checking for gcc -fnested-functions... no
checking for clang strsep/strcmp optimization... no
configure: checking OPENSSL with pkg-config
configure: checking whether system openssl > 1.1.0
checking for openssl >= 1.1.0... yes
checking for embedded pjproject (may have to download)... configuring
[pjproject]  Downloading https://raw.githubusercontent.com/asterisk/third-party/master/pjproject/2.16/pjproject-2.16.tar.bz2 to /tmp/pjproject-2.16.tar.bz2
[pjproject]  Verifying /tmp/pjproject-2.16.tar.bz2
[pjproject]  Verify successful
[pjproject]  Verifying /tmp/pjproject-2.16.tar.bz2
[pjproject]  Verify successful
[pjproject]  Unpacking /tmp/pjproject-2.16.tar.bz2
[pjproject]  Applying patches /build/asterisk/src/asterisk-23.3.0/third-party/pjproject/patches /build/asterisk/src/asterisk-23.3.0/third-party/pjproject/source
[pjproject]  Applying user.mak
[pjproject]  Rebuilding
[pjproject]  Applying custom include file patches/asterisk_malloc_debug.h
[pjproject]  Applying custom include file patches/config_site.h
[pjproject]  Configuring with --with-ssl --prefix=/opt/pjproject --disable-upnp --disable-speex-codec --disable-speex-aec --disable-bcg729 --disable-gsm-codec --disable-ilbc-codec --disable-l16-codec --disable-g722-codec --disable-g7221-codec --disable-opencore-amr --disable-silk --disable-opus --disable-video --disable-v4l2 --disable-sound --disable-ext-sound --disable-sdl --disable-libyuv --disable-ffmpeg --disable-openh264 --disable-ipp --disable-libwebrtc --disable-libsrtp --disable-upnp --without-external-pa --without-external-srtp --disable-resample --disable-g711-codec --enable-epoll
checking for bundled pjproject... yes
checking for gawk... (cached) gawk
checking for curl-config... /usr/bin/curl-config
checking for the version of libcurl... 8.20.0
checking for libcurl >= version 7.10.1... yes
checking whether libcurl is usable... yes
checking for curl_free... yes
checking for size_t... yes
checking for working alloca.h... yes
checking for alloca... yes
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for sys/wait.h that is POSIX.1 compatible... yes
checking for sys/types.h... (cached) yes
checking for netinet/in.h... yes
checking for arpa/nameser.h... yes
checking for netdb.h... yes
checking for resolv.h... yes
checking for arpa/nameser.h... (cached) yes
checking for assert.h... yes
checking for ctype.h... yes
checking for dlfcn.h... yes
checking for errno.h... yes
checking for fcntl.h... yes
checking for float.h... yes
checking for grp.h... yes
checking for inttypes.h... (cached) yes
checking for limits.h... yes
checking for locale.h... yes
checking for math.h... yes
checking for pwd.h... yes
checking for netinet/in.h... (cached) yes
checking for regex.h... yes
checking for sched.h... yes
checking for stdarg.h... yes
checking for stdint.h... (cached) yes
checking for stdio.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for syslog.h... yes
checking for sys/file.h... yes
checking for sys/ioctl.h... yes
checking for sys/param.h... (cached) yes
checking for sys/resource.h... yes
checking for sys/socket.h... (cached) yes
checking for sys/stat.h... (cached) yes
checking for sys/time.h... yes
checking for sys/types.h... (cached) yes
checking for sys/un.h... yes
checking for termios.h... yes
checking for time.h... yes
checking for unistd.h... (cached) yes
checking for arpa/inet.h... yes
checking for libintl.h... yes
checking for malloc.h... yes
checking for netdb.h... (cached) yes
checking for stddef.h... yes
checking for strings.h... (cached) yes
checking for sys/event.h... no
checking for utime.h... (cached) yes
checking for libedit... yes
checking for Testing for libedit unicode support... yes
checking for uuid_generate_random in -luuid... yes
checking for uuid/uuid.h... yes
checking for jansson >= 2.11... yes
checking for libjwt >= 1.15.3... no
checking for clock_gettime in -lrt... yes
checking for libxml-2.0... yes
checking for uriParseUriA in -luriparser... no
checking for xsltLoadStylesheetPI in -lxslt... yes
checking for libxslt/xsltInternals.h... yes
checking for xsltCleanupGlobals in -lxslt... yes
checking for libxslt/xsltInternals.h... (cached) yes
checking for xlocale.h... no
checking for winsock.h... no
checking for winsock2.h... no
checking for sys/poll.h... yes
checking for  option for large files... none needed
checking for _Bool... yes
checking for stdbool.h that conforms to C99 or later... yes
checking for an ANSI C-conforming const... yes
checking for uid_t... yes
checking for gid_t... yes
checking for inline... inline
checking for long double with more range or precision than double... yes
checking for mode_t... yes
checking for off_t... yes
checking for pid_t... yes
checking for size_t... (cached) yes
checking for struct stat.st_blksize... yes
checking for struct ucred.uid... yes
checking for struct ucred.cr_uid... no
checking for struct sockpeercred.uid... no
checking for struct ifreq.ifr_ifru.ifru_hwaddr... yes
checking whether struct tm is in sys/time.h or time.h... time.h
checking for working volatile... yes
checking for ptrdiff_t... yes
checking for struct stat.st_mtim... yes
checking for struct stat.st_mtimensec... no
checking for struct stat.st_mtimespec... no
checking for struct sockaddr_in6.sin6_len... no
checking for struct sockaddr_in.sin_len... no
checking for working chown... yes
checking whether closedir returns void... no
checking for error_at_line... yes
checking for vfork.h... no
checking for fork... yes
checking for vfork... yes
checking for working fork... yes
checking for working vfork... (cached) yes
checking for declarations of fseeko and ftello... yes
checking for working memcmp... yes
checking for getpagesize... yes
checking for vprintf... yes
checking for working mmap... yes
checking types of arguments for select... int,fd_set *,struct timeval *
checking whether lstat correctly handles trailing slash... yes
checking whether stat accepts an empty string... no
checking for working strcoll... yes
checking for strftime... yes
checking for working strnlen... yes
checking for working strtod... yes
checking whether utime accepts a null argument... yes
checking for asprintf... yes
checking for atexit... yes
checking for closefrom... yes
checking for dup2... yes
checking for eaccess... yes
checking for endpwent... yes
checking for euidaccess... yes
checking for ffsll... yes
checking for ftruncate... yes
checking for getcwd... yes
checking for gethostbyname... yes
checking for gethostname... yes
checking for getloadavg... yes
checking for gettimeofday... yes
checking for glob... yes
checking for ioperm... yes
checking for inet_ntoa... yes
checking for isascii... yes
checking for memchr... yes
checking for memmove... yes
checking for memset... yes
checking for mkdir... yes
checking for mkdtemp... yes
checking for munmap... yes
checking for newlocale... yes
checking for pipe2... yes
checking for ppoll... yes
checking for putenv... yes
checking for re_comp... yes
checking for regcomp... yes
checking for select... yes
checking for setenv... yes
checking for socket... yes
checking for strcasecmp... yes
checking for strcasestr... yes
checking for strchr... yes
checking for strcspn... yes
checking for strdup... yes
checking for strerror... yes
checking for strlcat... yes
checking for strlcpy... yes
checking for strncasecmp... yes
checking for strndup... yes
checking for strnlen... yes
checking for strrchr... yes
checking for strsep... yes
checking for strspn... yes
checking for strstr... yes
checking for strtod... (cached) yes
checking for strtol... yes
checking for strtold... yes
checking for strtoq... yes
checking for unsetenv... yes
checking for uselocale... yes
checking for utime... yes
checking for vasprintf... yes
checking for getpeereid... no
checking for sysctl... no
checking for swapctl... no
checking for malloc_trim... yes
checking for htonll... no
checking for ntohll... no
checking for sqrt in -lm... yes
checking for exp2... yes
checking for log2... yes
checking for exp10... yes
checking for log10... yes
checking for sin... yes
checking for cos... yes
checking for tan... yes
checking for asin... yes
checking for acos... yes
checking for atan... yes
checking for atan2... yes
checking for pow... yes
checking for rint... yes
checking for exp... yes
checking for log... yes
checking for remainder... yes
checking for fmod... yes
checking for round... yes
checking for roundf... yes
checking for trunc... yes
checking for floor... yes
checking for ceil... yes
checking for exp2l... yes
checking for log2l... yes
checking for exp10l... yes
checking for log10l... yes
checking for sinl... yes
checking for cosl... yes
checking for tanl... yes
checking for asinl... yes
checking for acosl... yes
checking for atanl... yes
checking for atan2l... yes
checking for powl... yes
checking for sqrtl... yes
checking for rintl... yes
checking for expl... yes
checking for logl... yes
checking for remainderl... yes
checking for fmodl... yes
checking for roundl... yes
checking for truncl... yes
checking for floorl... yes
checking for ceill... yes
checking for LLONG_MAX in limits.h... yes
checking for timersub in time.h... yes
checking for a version of GNU ld that supports the --dynamic-list flag... yes
checking for sys/poll.h... (cached) yes
checking for inet_aton... yes
checking for IP_PKTINFO... yes
checking for gcc options needed to detect all undeclared functions... none needed
checking whether gethostbyname_r is declared... yes
checking for library containing gethostbyname_r... none required
checking for gethostbyname_r with 6 arguments... yes
checking for gethostbyname_r with 5 arguments... no
checking for byteswap.h... yes
checking for __swap16 variant of <sys/endian.h> byteswapping macros... no
checking for bswap16 variant of <sys/endian.h> byteswapping macros... no
checking for locale_t in locale.h... yes
checking for O_EVTONLY in fcntl.h... no
checking for O_SYMLINK in fcntl.h... no
checking for PTHREAD_RWLOCK_INITIALIZER in pthread.h... yes
checking for PTHREAD_RWLOCK_PREFER_WRITER_NP in pthread.h... yes
checking for PTHREAD_MUTEX_RECURSIVE_NP in pthread.h... yes
checking for PTHREAD_MUTEX_ADAPTIVE_NP in pthread.h... yes
checking for pthread_spinlock_t in pthread.h... yes
checking for pthread_rwlock_timedwrlock() in pthread.h... yes
checking for working unnamed semaphores... yes
checking if PTHREAD_ONCE_INIT needs braces... no
checking for PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP in pthread.h... yes
checking whether we can compare a mutex to its initial value... no
checking for compiler sync operations... yes
checking for compiler atomic operations... yes
checking if your system printf is NULL-safe.... (null)yes
checking if socket() accepts SOCK_NONBLOCK... yes
checking if setsockopt() accepts the IPV6_V6ONLY socket option... yes
checking if we can increase the maximum select-able file descriptor... no
checking if we have usable eventfd support... yes
checking for compiler 'attribute pure' support... yes
checking for compiler 'attribute malloc' support... yes
checking for compiler 'attribute const' support... yes
checking for compiler 'attribute unused' support... yes
checking for compiler 'attribute always_inline' support... no
checking for compiler 'attribute deprecated' support... yes
checking for compiler 'attribute sentinel' support... yes
checking for compiler 'attribute warn_unused_result' support... yes
checking for compiler 'attribute may_alias' support... yes
checking for compiler 'attribute constructor' support... yes
checking for compiler 'attribute destructor' support... yes
checking for compiler 'attribute noreturn' support... yes
checking for -fsanitize=address support... yes
checking for -fsanitize=thread support... yes
checking for -fsanitize=leak support... yes
checking for -fsanitize=undefined support... yes
checking for -Wdeclaration-after-statement support... yes
checking for -Wtrampolines support... yes
checking for _FORTIFY_SOURCE support... yes
checking for -fno-strict-overflow... yes
checking for -Wno-format-truncation... yes
checking for -Wno-format-y2k... yes
checking for -Wno-stringop-truncation... yes
checking for -Wshadow... yes
checking for -march=native support... yes
checking whether to use rpath... not needed
checking for sysinfo... yes
checking for library containing res_9_ninit... no
checking for res_ninit... yes
checking for library containing res_9_ndestroy... no
checking for res_ndestroy... no
checking for library containing res_9_close... no
checking for res_close... no
checking for struct __res_state._u._ext.nsaddrs... yes
checking for BIND_8_COMPAT required... no
checking for GLOB_NOMAGIC in glob.h... yes
checking for GLOB_BRACE in glob.h... yes
checking for FNM_CASEFOLD in fnmatch.h... yes
checking for RTLD_NOLOAD in dlfcn.h... yes
checking for IP_MTU_DISCOVER in netinet/in.h... yes
checking size of int... 4
checking size of long... 8
checking size of long long... 8
checking size of char *... 8
checking size of long... (cached) 8
checking size of long long... (cached) 8
checking size of fd_set.fds_bits... (cached) 8
checking for dladdr in dlfcn.h... yes
checking for bfd_openr in -lbfd... yes
checking for bfd.h... yes
checking for cap_from_text in -lcap... yes
checking for sys/capability.h... yes
checking for DAHDI_RESET_COUNTERS in dahdi/user.h... no
checking for DAHDI_DEFAULT_MTU_MRU in dahdi/user.h... no
checking for DAHDI_CODE in dahdi/user.h... no
checking for DAHDI_POLICY_HALF_FULL in dahdi/user.h... no
checking for enhanced dahdi vmwi support... no
checking if "int foo = DAHDI_ECHOCANCEL_FAX_MODE" compiles using dahdi/user.h... no
checking for getifaddrs() support... yes
checking for timerfd support... yes
checking for gsm_create in -lgsm... yes
checking for gsm.h... no
checking for gsm/gsm.h... yes
checking for libilbc < 3... no
checking for iconv_open in -liconv... no
checking for libiconv_open in -liconv... no
checking for iconv_close in -lc... yes
checking for iconv.h... yes
checking for icaltimezone_get_utc_timezone in -lical... yes
checking for libical/ical.h... yes
checking for iks_start_sasl in -liksemel... no
checking for UW IMAP Toolkit c-client library... no
checking for system c-client library...... no
checking for SQLConnect in -liodbc... no
checking for inotify_init in -lc... yes
checking for sys/inotify.h... yes
checking for jack_activate in -ljack... no
checking for kqueue in -lc... no
checking for kevent64... no
checking for ldap_initialize in -lldap... yes
checking for ldap.h... yes
checking for mysql_config... no
checking for neon-config... /bin/neon-config
checking for neon-config... /bin/neon-config
checking for netsnmp-agent... no
checking for net-snmp-config... no
checking for newtBell in -lnewt... no
checking for ub_ctx_delete in -lunbound... no
checking for SQLConnect in -lodbc... no
checking for ogg_stream_init in -logg... yes
checking for ogg/ogg.h... yes
checking for backtrace in -lexecinfo... no
checking for backtrace in -lc... yes
checking for execinfo.h... yes
checking for ba2str in -lbluetooth... no
checking for bs_version in -lbeanstalk... no
checking for pg_config... no
checking for poptStrerror in -lpopt... yes
checking for popt.h... yes
checking for portaudio-2.0... no
checking for Pa_GetDeviceCount in -lportaudio... no
checking for pri_connected_line_update in -lpri... no
checking for resample_open in -lresample... no
checking for fftw_malloc in -lfftw3... no
checking for sf_open in -lsndfile... no
checking for minimum version of SpanDSP... no
checking for ss7_set_isup_timer in -lss7... no
checking for openr2_chan_new in -lopenr2... no
checking for opus_encoder_create in -lopus... yes
checking for opus/opus.h... yes
checking for op_open_callbacks in -lopusfile... no
checking for luaL_newstate in -llua5.4... no
checking for luaL_newstate in -llua5.3... no
checking for luaL_newstate in -llua5.2... no
checking for luaL_newstate in -llua5.1... no
checking for luaL_newstate in -llua... no
checking for rc_read_config in -lfreeradius-client... no
checking for rc_read_config in -lradiusclient-ng... no
checking for rc_read_config in -lradcli... no
checking for codec2_create in -lcodec2... no
checking for cpg_join in -lcpg... no
checking for corosync_cfg_state_track in -lcfg... no
checking for speex_encode in -lspeex... yes
checking for speex/speex.h... yes
checking for speex_preprocess_ctl in -lspeex... no
checking for speex_preprocess_ctl in -lspeexdsp... yes
checking for speex/speex.h... (cached) yes
checking for sqlite3_open in -lsqlite3... yes
checking for sqlite3.h... yes
checking for crypt in -lcrypt... yes
checking for crypt.h... yes
checking for crypt... no
checking for crypt_r in -lcrypt... yes
checking for srtp_init in -lsrtp2... yes
checking for srtp2/srtp.h... yes
checking for the ability of -lsrtp2 to be linked in a shared object... yes
checking for srtp_crypto_policy_set_aes_cm_256_hmac_sha1_80 in -lsrtp2... yes
checking for srtp_crypto_policy_set_aes_cm_192_hmac_sha1_80 in -lsrtp2... yes
checking for srtp_crypto_policy_set_aes_gcm_128_8_auth in -lsrtp2... yes
checking for srtp_shutdown in -lsrtp2... yes
checking for srtp2/srtp.h... (cached) yes
checking for srtp_get_version_string in -lsrtp2... yes
checking for srtp2/srtp.h... (cached) yes
checking for gmime-3.0... no
checking for gmime-2.6... no
checking for gmime-2.4... no
checking for gmime-2.2... no
checking for gmime-2.0... no
checking for malloc in -lhoard... no
checking for dbinit in -lsybdb... no
checking for tone_zone_find_by_num in -ltonezone... no
checking for tone_zone_find in -ltonezone... no
checking for vorbis_info_init in -lvorbis... yes
checking for vorbis/codec.h... yes
checking for OV_CALLBACKS_NOCLOSE declared in vorbis/vorbisfile.h... yes
checking for compress in -lz... yes
checking for zlib.h... yes
checking for linux/compiler.h... no
checking for sdl-config... no
checking for IMG_Load in -lSDL_image... no
checking for sws_getContext in -lavcodec... no
checking for linux/videodev.h... no
checking for XOpenDisplay in -lX11... no
checking for XOpenDisplay in -lX11... (cached) no
checking for /sbin/launchd... no
checking for gtk+-2.0... no
checking for libsystemd... yes
checking for LOG_AUTH in syslog.h... yes
checking for LOG_AUTHPRIV in syslog.h... yes
checking for LOG_CRON in syslog.h... yes
checking for LOG_DAEMON in syslog.h... yes
checking for LOG_FTP in syslog.h... yes
checking for LOG_KERN in syslog.h... yes
checking for LOG_LPR in syslog.h... yes
checking for LOG_MAIL in syslog.h... yes
checking for LOG_NEWS in syslog.h... yes
checking for LOG_SYSLOG in syslog.h... yes
checking for LOG_UUCP in syslog.h... yes
checking for bridges/bridge_softmix/include/hrirs.h... yes
checking for mandatory modules:  PJPROJECT... ok
configure: creating ./config.status
config.status: creating build_tools/menuselect-deps
config.status: creating makeopts
config.status: creating include/asterisk/autoconfig.h
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking for GNU make... make
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for size_t... yes
checking for working alloca.h... yes
checking for alloca... yes
checking for asprintf... yes
checking for getloadavg... yes
checking for setenv... yes
checking for strcasestr... yes
checking for strndup... yes
checking for strnlen... yes
checking for strsep... yes
checking for unsetenv... yes
checking for vasprintf... yes
checking for newtBell in -lnewt... no
checking for initscr in -lncurses... yes
checking for ncurses.h... yes
checking for keypad in -ltinfo... yes
checking for ncurses.h... (cached) yes
checking for a sed that does not truncate output... /usr/bin/sed
checking for libxml-2.0... yes
checking for gtk+-2.0... no
configure: creating ./config.status
config.status: creating makeopts
config.status: creating autoconfig.h
configure: Menuselect build configuration successfully completed

               .$$$$$$$$$$$$$$$=..
            .$7$7..          .7$$7:.
          .$$:.                 ,$7.7
        .$7.     7$$$$           .$$77
     ..$$.       $$$$$            .$$$7
    ..7$   .?.   $$$$$   .?.       7$$$.
   $.$.   .$$$7. $$$$7 .7$$$.      .$$$.
 .777.   .$$$$$$77$$$77$$$$$7.      $$$,
 $$$~      .7$$$$$$$$$$$$$7.       .$$$.
.$$7          .7$$$$$$$7:          ?$$$.
$$$          ?7$$$$$$$$$$I        .$$$7
$$$       .7$$$$$$$$$$$$$$$$      :$$$.
$$$       $$$$$$7$$$$$$$$$$$$    .$$$.
$$$        $$$   7$$$7  .$$$    .$$$.
$$$$             $$$$7         .$$$.
7$$$7            7$$$$        7$$$
 $$$$$                        $$$
  $$$$7.                       $$  (TM)
   $$$$$$$.           .7$$$$$$  $$
     $$$$$$$$$$$$7$$$$$$$$$.$$$$$$
       $$$$$$$$$$$$$$$$.

configure: Package configured for:
configure: OS type  : linux-gnu
configure: Host CPU : x86_64
configure: build-cpu:vendor:os: x86_64 : pc : linux-gnu :
configure: host-cpu:vendor:os: x86_64 : pc : linux-gnu :
make[1]: Entering directory '/build/asterisk/src/asterisk-23.3.0'
make[2]: Entering directory '/build/asterisk/src/asterisk-23.3.0/utils'
<rm -rf removed>
(cd source ; ./aconfigure  --with-ssl --prefix=/opt/pjproject --disable-upnp --disable-speex-codec --disable-speex-aec --disable-bcg729 --disable-gsm-codec --disable-ilbc-codec --disable-l16-codec --disable-g722-codec --disable-g7221-codec --disable-opencore-amr --disable-silk --disable-opus --disable-video --disable-v4l2 --disable-sound --disable-ext-sound --disable-sdl --disable-libyuv --disable-ffmpeg --disable-openh264 --disable-ipp --disable-libwebrtc --disable-libsrtp --disable-upnp --without-external-pa --without-external-srtp --disable-resample --disable-g711-codec --enable-epoll)
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether the compiler supports GNU C++... yes
checking whether g++ -std=gnu++11 accepts -g... yes
checking for g++ -std=gnu++11 option to enable C++11 features... none needed
checking for ranlib... ranlib
checking for pthread_create in -lpthread... yes
checking for puts in -lwsock32... no
checking for puts in -lws2_32... no
checking for puts in -lole32... no
checking for puts in -lwinmm... no
checking for puts in -lsocket... no
checking for puts in -lrt... yes
checking for sin in -lm... yes
checking for x86_64-linux-gnu-pkg-config... no
checking for x86_64-pc-linux-gnu-pkg-config... x86_64-pc-linux-gnu-pkg-config
checking for uuid_generate in -luuid... yes
checking for uuid_generate in -luuid... (cached) yes
checking for library containing gethostbyname... none required
Setting PJ_M_NAME to x86_64
checking memory alignment... 8 bytes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking whether byte ordering is bigendian... no
Checking if floating point is disabled... no
checking for arpa/inet.h... yes
checking for assert.h... yes
checking for ctype.h... yes
checking for errno.h... yes
checking for fcntl.h... yes
checking for linux/socket.h... yes
checking for limits.h... yes
checking for malloc.h... yes
checking for netdb.h... yes
checking for netinet/in_systm.h... yes
checking for netinet/in.h... yes
checking for netinet/ip.h... yes
checking for netinet/tcp.h... yes
checking for ifaddrs.h... yes
checking for getifaddrs... yes
checking for semaphore.h... yes
checking for setjmp.h... yes
checking for stdarg.h... yes
checking for stddef.h... yes
checking for stdio.h... (cached) yes
checking for stdint.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for inttypes.h... (cached) yes
checking for sys/ioctl.h... yes
checking for sys/select.h... yes
checking for sys/socket.h... yes
checking for sys/time.h... yes
checking for sys/timeb.h... yes
checking for sys/types.h... (cached) yes
checking for sys/filio.h... no
checking for sys/sockio.h... no
checking for sys/utsname.h... yes
checking for time.h... yes
checking for unistd.h... (cached) yes
checking for execinfo.h... yes
checking for winsock.h... no
checking for winsock2.h... no
checking for mswsock.h... no
checking for ws2tcpip.h... no
checking for uuid/uuid.h... yes
checking for net/if.h... yes
checking for localtime_r... yes
Setting PJ_OS_NAME to x86_64-pc-linux-gnu
Setting PJ_HAS_ERRNO_VAR to 1
Setting PJ_HAS_HIGH_RES_TIMER to 1
Setting PJ_HAS_MALLOC to 1
Setting PJ_NATIVE_STRING_IS_UNICODE to 0
Setting PJ_ATOMIC_VALUE_TYPE to long
checking if inet_aton() is available... yes
checking if inet_pton() is available... yes
checking if inet_ntop() is available... yes
checking if getaddrinfo() is available... yes
checking if socketpair() is available... yes
checking if sockaddr_in has sin_len member... no
checking if socklen_t is available... yes
checking if IPV6_V6ONLY is available... yes
checking if SO_ERROR is available... yes
checking if pthread_rwlock_t is available... yes
checking if pthread_mutexattr_settype() is available... no
checking if pthread_mutexattr_t has recursive member... no
checking ioqueue backend... /dev/epoll
Building shared libraries... no
Building pjsua2 library and application... yes
Checking if UPnP is disabled... yes
Checking if resampling is disabled...yes
Checking if sound is disabled... yes
checking for sys/soundcard.h... yes
checking for linux/soundcard.h... yes
checking for machine/soundcard.h... no
Video is disabled
Checking if small filter is disabled... no
Checking if large filter is disabled... no
Checking if Speex AEC is disabled...yes
Checking if G.711 codec is disabled...yes
Checking if L16 codecs are disabled...yes
Checking if GSM codec is disabled...yes
Checking if G.722 codec is disabled...yes
Checking if G.722.1 codec is disabled...yes
Checking if Speex codec is disabled...yes
Checking if iLBC codec is disabled...yes
Checking if libsamplerate is enabled...no
Building libresample as shared library... no
Checking if Speex resample is enabled... no
Checking if SDL is disabled... yes
Checking if ffmpeg is disabled... yes
Checking if V4L2 is disabled... yes
Checking if OpenH264 is disabled... yes
Checking if VPX is disabled... yes
Skipping Intel IPP settings (not wanted)
Using SSL prefix... yes
checking for OpenSSL installations..
checking for openssl/ssl.h... yes
checking for ERR_load_BIO_strings in -lcrypto... yes
checking for SSL_CTX_new in -lssl... yes
OpenSSL library found, SSL support enabled
checking for EVP_aes_128_gcm in -lcrypto... yes
OpenSSL has AES GCM support, SRTP will use OpenSSL
Checking if OpenCORE AMR support is disabled... yes
Checking if SILK support is disabled... yes
Checking if OPUS support is disabled... yes
Checking if bcg729 is disabled... yes
checking lyra usability... no
Checking if libsrtp is disabled...yes
Checking if libyuv is disabled...yes
Checking if libwebrtc is disabled...yes
Checking if libwebrtc-aec3 is enabled...no
checking if select() needs correct nfds... no (default)
** Decided that select() doesn't need correct nfds (please check)
checking if pj_thread_create() should enforce stack size... no (default)
checking if pj_thread_create() should allocate stack... no (default)
** Setting non-blocking recv() retval to EAGAIN (please check)
** Setting non-blocking connect() retval to EINPROGRESS (please check)
checking for pthread_np.h... no
checking for pthread_setname_np... yes
checking for pthread_set_name_np... no
aconfigure: creating ./config.status
config.status: creating build.mak
config.status: creating build/os-auto.mak
config.status: creating build/cc-auto.mak
config.status: creating pjlib/build/os-auto.mak
config.status: creating pjlib-util/build/os-auto.mak
config.status: creating pjmedia/build/os-auto.mak
config.status: creating pjsip/build/os-auto.mak
config.status: creating third_party/build/os-auto.mak
config.status: creating pjlib/include/pj/compat/os_auto.h
config.status: pjlib/include/pj/compat/os_auto.h is unchanged
config.status: creating pjlib/include/pj/compat/m_auto.h
config.status: pjlib/include/pj/compat/m_auto.h is unchanged
config.status: creating pjmedia/include/pjmedia/config_auto.h
config.status: pjmedia/include/pjmedia/config_auto.h is unchanged
config.status: creating pjmedia/include/pjmedia-codec/config_auto.h
config.status: pjmedia/include/pjmedia-codec/config_auto.h is unchanged
config.status: creating pjsip/include/pjsip/sip_autoconf.h
config.status: pjsip/include/pjsip/sip_autoconf.h is unchanged


Configurations for current target have been written to 'build.mak', and 'os-auto.mak' in various build directories, and pjlib/include/pj/compat/os_auto.h.

Further customizations can be put in:
  - 'user.mak'
  - 'pjlib/include/pj/config_site.h'

The next step now is to run 'make dep' and 'make'.

<Hopefully irrelevant log output removed>

gcc -o ari/resource_sounds.o -c ari/resource_sounds.c -MD -MT ari/resource_sounds.o -MF .ari_resource_sounds.o.d -MP -pthread -I/build/asterisk/src/asterisk-23.3.0/include -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/asterisk/src=/usr/src/debug/asterisk -flto=auto  -I/usr/include/libxml2 -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -fPIC -DAST_MODULE=\"res_ari_sounds\" -DAST_MODULE_SELF_SYM=__internal_res_ari_sounds_self
<command-line>: warning: ‘_FORTIFY_SOURCE’ redefined
<command-line>: note: this is the location of the previous definition
/build/asterisk/src/asterisk-23.3.0/build_tools/make_linker_version_script res_ari_sounds "" "/build/asterisk/src/asterisk-23.3.0"
gcc -o res_ari_sounds.so -pthread -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now          -Wl,-z,pack-relative-relocs -flto=auto  -shared -Wl,--version-script,res_ari_sounds.exports,--warn-common  res_ari_sounds.o ari/resource_sounds.o -lpthread
gcc -o res_audiosocket.o -c res_audiosocket.c -MD -MT res_audiosocket.o -MF .res_audiosocket.o.d -MP -pthread -I/build/asterisk/src/asterisk-23.3.0/include -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/asterisk/src=/usr/src/debug/asterisk -flto=auto  -I/usr/include/libxml2 -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -fPIC -DAST_MODULE=\"res_audiosocket\" -DAST_MODULE_SELF_SYM=__internal_res_audiosocket_self
<command-line>: warning: ‘_FORTIFY_SOURCE’ redefined
<command-line>: note: this is the location of the previous definition
/build/asterisk/src/asterisk-23.3.0/build_tools/make_linker_version_script res_audiosocket "" "/build/asterisk/src/asterisk-23.3.0"
gcc -o res_audiosocket.so -pthread -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now          -Wl,-z,pack-relative-relocs -flto=auto  -shared -Wl,--version-script,res_audiosocket.exports,--warn-common  res_audiosocket.o -lpthread
gcc -o res_calendar.o -c res_calendar.c -MD -MT res_calendar.o -MF .res_calendar.o.d -MP -pthread -I/build/asterisk/src/asterisk-23.3.0/include -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/asterisk/src=/usr/src/debug/asterisk -flto=auto  -I/usr/include/libxml2 -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -fPIC -DAST_MODULE=\"res_calendar\" -DAST_MODULE_SELF_SYM=__internal_res_calendar_self
<command-line>: warning: ‘_FORTIFY_SOURCE’ redefined
<command-line>: note: this is the location of the previous definition
/build/asterisk/src/asterisk-23.3.0/build_tools/make_linker_version_script res_calendar "" "/build/asterisk/src/asterisk-23.3.0"
gcc -o res_calendar.so -pthread -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now          -Wl,-z,pack-relative-relocs -flto=auto  -shared -Wl,--version-script,res_calendar.exports,--warn-common  res_calendar.o -lpthread
gcc -o res_calendar_caldav.o -c res_calendar_caldav.c -MD -MT res_calendar_caldav.o -MF .res_calendar_caldav.o.d -MP -pthread -I/build/asterisk/src/asterisk-23.3.0/include -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/asterisk/src=/usr/src/debug/asterisk -flto=auto  -I/usr/include/libxml2 -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -fPIC -DAST_MODULE=\"res_calendar_caldav\" -DAST_MODULE_SELF_SYM=__internal_res_calendar_caldav_self  -I/usr/include/neon -pthread -I/usr/include/libxml2
<command-line>: warning: ‘_FORTIFY_SOURCE’ redefined
<command-line>: note: this is the location of the previous definition
res_calendar_caldav.c: In function ‘caldav_add_event’:
res_calendar_caldav.c:467:23: error: implicit declaration of function ‘icaltime_add’; did you mean ‘icaltime_t’? [-Wimplicit-function-declaration]
  467 |                 tmp = icaltime_add(start, trigger.duration);
      |                       ^~~~~~~~~~~~
      |                       icaltime_t
res_calendar_caldav.c:467:23: error: incompatible types when assigning to type ‘icaltimetype’ from type ‘int’
res_calendar_caldav.c: In function ‘handle_end_element’:
res_calendar_caldav.c:528:68: error: passing argument 4 of ‘icalcomponent_foreach_recurrence’ from incompatible pointer type [-Wincompatible-pointer-types]
  528 |                 icalcomponent_foreach_recurrence(iter, start, end, caldav_add_event, state->pvt);
      |                                                                    ^~~~~~~~~~~~~~~~
      |                                                                    |
      |                                                                    void (*)(icalcomponent *, struct icaltime_span *, void *) {aka void (*)(struct icalcomponent_impl *, struct icaltime_span *, void *)}
In file included from res_calendar_caldav.c:33:
/usr/include/libical/ical.h:7973:66: note: expected ‘void (*)(icalcomponent *, const struct icaltime_span *, void *)’ {aka ‘void (*)(struct icalcomponent_impl *, const struct icaltime_span *, void *)’} but argument is of type ‘void (*)(icalcomponent *, struct icaltime_span *, void *)’ {aka ‘void (*)(struct icalcomponent_impl *, struct icaltime_span *, void *)’}
 7973 |                                                           void (*callback)(icalcomponent *comp,
      |                                                           ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 7974 |                                                                            const struct icaltime_span *span,
      |                                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 7975 |                                                                            void *data),
      |                                                                            ~~~~~~~~~~~
res_calendar_caldav.c:352:13: note: ‘caldav_add_event’ declared here
  352 | static void caldav_add_event(icalcomponent *comp, struct icaltime_span *span, void *data)
      |             ^~~~~~~~~~~~~~~~
make[1]: *** [/build/asterisk/src/asterisk-23.3.0/Makefile.rules:172: res_calendar_caldav.o] Error 1
make[1]: Leaving directory '/build/asterisk/src/asterisk-23.3.0/res'
make: *** [Makefile:402: res] Error 2
```

### Asterisk Issue Guidelines

- [x] Yes, I have read the Asterisk Issue Guidelines

--- a/res/res_calendar_caldav.c
+++ b/res/res_calendar_caldav.c
@@ -31,6 +31,12 @@
 #include "asterisk.h"
 
 #include <libical/ical.h>
+
+#if ICAL_MAJOR_VERSION >= 4
+#define ICAL_SPAN_CONST const
+#else
+#define ICAL_SPAN_CONST
+#endif
 #include <ne_session.h>
 #include <ne_uri.h>
 #include <ne_request.h>
@@ -349,7 +355,7 @@ static time_t icalfloat_to_timet(icaltimetype time)
  * span here, and instead will grab the start and end from the component, which will
  * allow us to test for floating times or dates.
  */
-static void caldav_add_event(icalcomponent *comp, struct icaltime_span *span, void *data)
+static void caldav_add_event(icalcomponent *comp, ICAL_SPAN_CONST struct icaltime_span *span, void *data)
 {
 	struct caldav_pvt *pvt = data;
 	struct ast_calendar_event *event;
@@ -464,7 +470,17 @@ static void caldav_add_event(icalcomponent *comp, struct icaltime_span *span, vo
 	} else { /* Offset from either dtstart or dtend */
 		/* XXX Technically you can check RELATED to see if the event fires from the END of the event
 		 * But, I'm not sure I've ever seen anyone implement it in calendaring software, so I'm ignoring for now */
+#if ICAL_MAJOR_VERSION >= 4
+		tmp = start;
+		int sign = trigger.duration.is_neg ? -1 : 1;
+		icaltime_adjust(&tmp,
+			sign * (trigger.duration.days + trigger.duration.weeks * 7),
+			sign * trigger.duration.hours,
+			sign * trigger.duration.minutes,
+			sign * trigger.duration.seconds);
+#else
 		tmp = icaltime_add(start, trigger.duration);
+#endif
 		event->alarm = icaltime_as_timet_with_zone(tmp, icaltime_get_timezone(start));
 	}
 

--- a/res/res_calendar_icalendar.c
+++ b/res/res_calendar_icalendar.c
@@ -30,6 +30,12 @@
 #include "asterisk.h"
 
 #include <libical/ical.h>
+
+#if ICAL_MAJOR_VERSION >= 4
+#define ICAL_SPAN_CONST const
+#else
+#define ICAL_SPAN_CONST
+#endif
 #include <ne_session.h>
 #include <ne_uri.h>
 #include <ne_request.h>
@@ -191,7 +197,7 @@ static time_t icalfloat_to_timet(icaltimetype time)
  * allow us to test for floating times or dates.
  */
 
-static void icalendar_add_event(icalcomponent *comp, struct icaltime_span *span, void *data)
+static void icalendar_add_event(icalcomponent *comp, ICAL_SPAN_CONST struct icaltime_span *span, void *data)
 {
 	struct icalendar_pvt *pvt = data;
 	struct ast_calendar_event *event;
@@ -341,7 +347,17 @@ static void icalendar_add_event(icalcomponent *comp, struct icaltime_span *span,
 	} else { /* Offset from either dtstart or dtend */
 		/* XXX Technically you can check RELATED to see if the event fires from the END of the event
 		 * But, I'm not sure I've ever seen anyone implement it in calendaring software, so I'm ignoring for now */
+#if ICAL_MAJOR_VERSION >= 4
+		tmp = start;
+		int sign = trigger.duration.is_neg ? -1 : 1;
+		icaltime_adjust(&tmp,
+			sign * (trigger.duration.days + trigger.duration.weeks * 7),
+			sign * trigger.duration.hours,
+			sign * trigger.duration.minutes,
+			sign * trigger.duration.seconds);
+#else
 		tmp = icaltime_add(start, trigger.duration);
+#endif
 		event->alarm = icaltime_as_timet_with_zone(tmp, icaltime_get_timezone(start));
 	}
 

